File: Linguagem C/diagonal_negativos.cpp
#include <stdio.h>

int main(void){
	
	int N;
	int negativo = 0;
	
	printf("Qual a ordem da matriz? ");
	scanf("%d", &N);
	
	int vetor[N][N];

	
	for(int i=0; i < N; i++) {
		for(int j=0; j < N; j++) {
			
			printf("Elemento [%d,%d]: ", i, j);
			scanf("%d", &vetor[i][j]);
		}
	}
	
	for(int i=0; i < N; i++){
		for(int j=0; j < N; j++){
			
			if(vetor[i][j] < 0) {
				negativo++;
			}	
		}
	}
	
	printf("DIAGONAL PRINCIPAL:\n");
	for(int i=0; i < N; i++){
		printf("%d ", vetor[i][i]);
	}
	printf("\nQUANTIDADE DE NEGATIVOS: %d", negativo);
	
return 0;		
}

File: Python/diagonal_negativos.py
'''Fazer um programa para ler um número inteiro N (máximo = 10) e uma matriz quadrada de ordem N contendo números inteiros. 
Em seguida, mostrar a diagonal principal e a quantidade de valores negativos da matriz.'''

quantNegativos = int(0)

N = int(input("Qual a ordem da matriz? "))

matriz = [[0 for x in range(N)] for x in range(N)]

#Inserindo valores a matriz
for i in range(0, N):
    for j in range(0, N):
        matriz[i][j] = int(input(f"Elemento [{i},{j}]: "))


#Percorrendo a matriz e achando a diagonal
print("Diagonal principal:")

for i in range(0, N):
    print(f"{matriz[i][i]} ", end="")

print()

#Percorrendo a matriz para contar os numeros negativos.
for i in range(0, N):
    for j in range(0, N):
        if matriz[i][j] < 0:
            quantNegativos += 1

print(f"Quantidade de negativos: {quantNegativos}")